scapy/sendrecv.py

```diff
@@ -868,22 +869,28 @@ def _run(self,
                     tcpdump(fname, args=["-w", "-", flt], getfd=True)
                 ), label) for fname, label in six.iteritems(offline))
             else:
-                # Write Scapy Packet objects to a pcap file
-                def _write_to_pcap(packets_list):
+                def _write_to_pcap(packets):
                     filename = get_temp_file(autoext=".pcap")
-                    wrpcap(filename, offline)
-                    return filename, filename
-
-                if isinstance(offline, Packet):
-                    tempfile_written, offline = _write_to_pcap([offline])
-                elif isinstance(offline, list) and \
-                        all(isinstance(elt, Packet) for elt in offline):
-                    tempfile_written, offline = _write_to_pcap(offline)
+                    wrpcap(filename, packets)
+                    return filename
+
+                if isinstance(offline, Packet) or \
+                    isinstance(offline, PacketList) or \
+                    (isinstance(offline, list) and
+                     all(isinstance(elt, Packet) for elt in offline)):
```

Comment:
If it's a list of something else than packets what happens? If the answer is "it crashes" then this line is unnecessary

Reply:
Hey @gpotter2 

It doesn't crash, though the result can be considered "weird" - (The counter of `Other` will be equal to the list's length). I guess the right thing to do is to raise a TypeError or something.. I just copied the same type checks from the previous version to my changes.